src/main/java/store/controller/StoreController.java

```diff
@@ -0,0 +1,118 @@
+package store.controller;
+
+import store.Message.ErrorMessage;
+import store.model.domain.ShoppingCartCheck;
+import store.model.ReceiptTotals;
+import store.model.service.StoreService;
+import store.model.service.ReceiptCalculationService;
+import store.util.StringParser;
+import store.view.InputView;
+import store.view.OutputView;
+
+import java.util.List;
+import java.util.Map;
+
+public class StoreController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final StoreService storeService;
+    private final ReceiptCalculationService receiptCalculationService;
+
+    public StoreController(InputView inputView, OutputView outputView, StoreService storeService, ReceiptCalculationService receiptCalculationService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.storeService = storeService;
+        this.receiptCalculationService = receiptCalculationService;
+    }
+
+    public void run() {
+        do {
+            initializeView();
+            Map<String, Integer> shoppingCart = askPurchase();
+            List<ShoppingCartCheck> shoppingCartChecks = processShoppingCart(shoppingCart);
+            boolean isMembership = processMembership();
+
+            ReceiptTotals totals = receiptCalculationService.calculateTotals(shoppingCartChecks, isMembership);
+            outputView.printReceipt(shoppingCartChecks, totals);
+
+            storeService.reduceProductStocks(shoppingCartChecks);
+        } while (askAdditionalPurchase());
+    }
+
+    private void initializeView() {
```

Comment:
initializeView() 라는 메서드는 뭔가 의문스럽습니다. 해당 메서드의 행동은 View 객체의 초기화가 아니라, 환영 문구 출력 등의 행동 아닌가요? 초기 출력이라는 점에서 initialize와 아예 관련 없진 않지만, 조금 메서드 시그니처에 출력이라는 것이 명시되어 있으면 좋을 것 같습니다. 

```suggestion
    private void printInitMessage() { }
```

Reply:
어떤 부분에서 의문을 느끼셨는지 알겠네요. 제가 view라고 표현한 것이 문제같습니다. 초기 화면 출력을 올바르지 못한 메서드 명명으로 처리한 것 같습니다. display로 시작하는 메서드 명을 사용하는 게 좋았겠어요.